core/Network/TLS/Types.hs

```diff
@@ -35,8 +43,26 @@ data SessionData = SessionData
     , sessionCompression :: CompressionID
     , sessionClientSNI   :: Maybe HostName
     , sessionSecret      :: ByteString
+    , sessionGroup       :: Maybe Group
+    , sessionTicketInfo  :: Maybe TLS13TicketInfo
+    , sessionALPN        :: Maybe ByteString
+    , sessionMaxEarlyDataSize :: Word32
```

Comment:
Now `Int` is used.

Reply:
If `Int` is 32 bits, how should we treat valid values of 2^31 or larger?

Reply:
Is this validation code acceptable?

```haskell
-- Word32 is used in TLS 1.3 protocol.
-- Int is used for API for Haskell TLS because it is natural.
-- If Int is 64 bits, users can specify bigger number than Word32.
-- If Int is 32 bits, 2^31 or larger may be converted into minus numbers.
safeNonNegative32 :: (Num a, Ord a) => a -> a
safeNonNegative32 x
  | x <= 0    = 0
  | otherwise = x
          `min` fromIntegral (maxBound :: Int)    -- Int is 32 bits
          `min` fromIntegral (maxBound :: Word32) -- Int is 64 bits
```

Reply:
I think `fromIntegral (maxBound :: Word32)` will be -1 with a 32-bit Int.
What about this instead:
```haskell
safeNonNegative32 :: Int -> Word32
safeNonNegative32 x
    | x <= 0                = 0
    | finiteBitSize x <= 32 = fromIntegral x
    | otherwise             =
        fromIntegral (x `min` fromIntegral (maxBound :: Word32))
```

Reply:
Nice!